### PR TITLE
Added documentation and code examples for W0134 return-in-finally checker

### PR DESCRIPTION
**Pull request by MariaMa-GitHub**
## Motivation and Context

W0134 return-in-finally is a new check in Pylint which was added at the end of last semester, so it needed to be added to the PythonTA documentation. This pull request resolves the issue by providing documentation and examples for this checker.

Fixes pyta-uoft/pyta#999.

## Your Changes

**Description**:

- Added documentation for W0134 return-in-finally checker.
- Wrote a code example that raises the W0134 return-in-finally error.
- Provided a correction to the example code that demonstrates how to resolve the W0134 return-in-finally error.

**Type of change** (select all that apply):

- [X] Documentation update (change that modifies or updates documentation only)

## Testing

This pull request was tested using PythonTA with the code below. 

```python
import python_ta

python_ta.check_all()
```
The PythonTA report correctly showed that the W0134 (return-in-finally) error was raised.

## Questions and Comments (if applicable)

I am not too sure how to classify the W0134 (return-in-finally) error. I chose to put it under "Improper Python usage" since I think that it does "indicate a misuse of variables, control flow, or other Python features in our code" and it is similar to other errors in the same section (such as Return outside function (E0104) and Unreachable (W0101)). However, when I ran PythonTA, it put the W0134 (return-in-finally) error under "Style and Convention", which also makes a bit of sense. Since I am not too sure how exactly this checker should be classified, I would like to ask under which section should it be put and in the future how should I decide if there are conflicts like this happening.

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have verified that the CI tests have passed. 
- [X] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

**Review by sanrodarmawan98 — APPROVED** [review]

**Review by sanrodarmawan98 — APPROVED** [review]

**Review by sanrodarmawan98 — APPROVED** [review]

File: docs/checkers/index.md [hunk 1]
```diff
@@ -488,7 +513,7 @@ This error occurs when a non-mapping value is used in a place where mapping is e
 
 ### Unnecessary negation (C0117)
 
-This error occurs when a boolean expression contains an nnecessary negation. If we are getting this
+This error occurs when a boolean expression contains an unnecessary negation. If we are getting this
```
Comment (david-yz-liu) [inline docs/checkers/index.md:516]: This change is not related to this pull request, and so should not be included here.
 
However, this is a good catch, so what you could do it make a second, separate pull request that just fixes this issue!

File: docs/checkers/index.md [hunk 2]
```diff
@@ -9,6 +9,31 @@ email to \[david at cs dot toronto dot edu\].
...
+
+```python
+def error_code():
+
+    error_codes = [0, 1, 2]
+
+    try:
+        return error_codes[2]
+    except IndexError:
+        ...
```
Comment (david-yz-liu) [inline docs/checkers/index.md:32]: I would modify this example to capture a more typical "intended" behaviour as follows:

- Make `error_codes` a function parameter (in both the correct version and original example)
- Keep the return statement in the `except`
- Remove the return statement in the `try` (perhaps replace it with a `print` or something), and instead make the bottom return statement `return error_codes[2]`, which is what the "typical" behaviour would be.

**Review by david-yz-liu — COMMENTED** [review]
Good work, @MariaMa-GitHub! I left a few inline comments, and in addition to that have two overall comments:

1. Since this is your first pull request, please add your name to the list of contributors :)
2. In terms of decided where to put the error, it's a bit ad hoc right now. One guiding principle I use is to look for similar errors, or errors that are related to the same language features. In this case, you should look for errors related to exceptions (some recent ones are the "too broad exception" ones) and put this entry close to those. Don't worry about categorizing as "style" vs. "error" within PythonTA though, that's something we can revisit later.

File: docs/checkers/index.md (hunk 2 above)
Comment (MariaMa-GitHub) [reply docs/checkers/index.md]: I just made the modifications and pushed the changes.

File: docs/checkers/index.md [hunk 3]
```diff
@@ -9,6 +9,29 @@ email to \[david at cs dot toronto dot edu\].
 These errors generally indicate a misuse of variables, control flow, or other Python features in our
 code.
 
+(W0134)=
+
+### Return in finally
+
+This error occurs when a `return` statement is used inside a `finally` block.
```
Comment (david-yz-liu) [inline docs/checkers/index.md:16]: It would be good to go into more detail about _why_ this is a problem (remember that the intended audience is students learning Python). You gave a great explanation in yesterday's meeting, and could write something similar here.

It's useful to use concrete examples to help with this. So, I would recommend that you refer to the example you created, e.g. "In the example below, ..."

File: docs/checkers/index.md (hunk 1 above)
Comment (MariaMa-GitHub) [reply docs/checkers/index.md]: It was clear, I just haven't had the change to do so. Now it is fixed and a separate pull request has been created.

File: docs/checkers/index.md [hunk 4]
```diff
@@ -9,6 +9,34 @@ email to \[david at cs dot toronto dot edu\].
 These errors generally indicate a misuse of variables, control flow, or other Python features in our
 code.
 
+(W0134)=
+
+### Return in finally
```
Comment (david-yz-liu) [inline docs/checkers/index.md:14]: Oh, I missed this last time but (1) you should add the error code at the send of the heading (to be consistent with other headings), and (2) make sure this is categorized under "Exceptions" to make it go with the other exception-related errors.

**Review by david-yz-liu — APPROVED** [review]
